tensorly/utils/jointdiag.py

```diff
@@ -0,0 +1,182 @@
+from itertools import combinations
+import tensorly as tl
+
+# Authors: Aaron Meyer <a@ameyer.me>
+#          Sebastian Seljak
+
+# License: BSD 3 clause
+
+
+def jointdiag(
+    X,
```

Comment:
I'd name this `matrices` or `matrices_tensor` - X makes me think it's a single matrix at first glance (though granted should read the docstring).